x/x.go

```diff
@@ -279,7 +279,7 @@ func ValidateAddress(addr string) bool {
 	if p, err := strconv.Atoi(port); err != nil || p <= 0 || p >= 65536 {
 		return false
 	}
-	if err := net.ParseIP(host); err == nil {
+	if ip := net.ParseIP(host); ip != nil {
```

Comment:
ParseIP returns nil if the IP address is invalid. See https://golang.org/pkg/net/#ParseIP